clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/ResizeAsgAtomicOperationUnitSpec.groovy

```diff
@@ -110,8 +109,38 @@ class ResizeAsgAtomicOperationUnitSpec extends Specification {
     capacity(0, 2, 3)    | 1          | 2          | 3              || true
   }
 
+  void "should support resize requests with only some fields to update"() {
+    def mockAutoScaling = Mock(AmazonAutoScaling)
+    def mockAmazonClientProvider = Mock(AmazonClientProvider)
+    mockAmazonClientProvider.getAutoScaling(_, _, true) >> mockAutoScaling
+    mockAutoScaling.describeAutoScalingGroups(_) >> new DescribeAutoScalingGroupsResult().withAutoScalingGroups(
+      new AutoScalingGroup().withAutoScalingGroupName("myasg-stack-v0001")
+    )
+
+    def description = new ResizeAsgDescription(
+      asgs: [[
+               serverGroupName: "myasg-stack-v000",
+               region         : "us-west-1",
+               capacity       : new ServerGroup.Capacity(min: 42) // no max or desired specified
```

Comment:
I would suggest parameterizing this capacity and then using a `where:` to pass in a variety of combinations

```
sourceMin | sourceMax | sourceDesired || expectedMin || expectedMax || expectedDesired
42        | null      | null          || 42          || null        || null
42        | 40        | null          || 42          || null        || null

... etc.
```

The split between inputs and outputs might not be necessary (since they're expected to be identical) but you get the point.

Motivation being that we're actually asserting that it works for all combinations.

Reply:
good idea. What do you think should happen when we have `(min: null, max: null, desired: null)`?